package/com.unity.formats.usd/Runtime/Scripts/IO/Geometry/MeshImporter.cs

```diff
@@ -266,67 +268,84 @@ public class GeometrySubsets {
       Material mat = renderer.sharedMaterial;
       bool changeHandedness = options.changeHandedness == BasisTransformation.SlowAndSafe;
 
+      //
+      // Purpose.
+      //
+
+      // Deactivate non-geometry prims (e.g. guides, render, etc).
+      if (usdMesh.purpose != Purpose.Default) {
+        go.SetActive(false);
+      }
+      
       //
       // Points.
       //
 
-      if (options.meshOptions.points == ImportMode.Import && usdMesh.points != null) {
-        if (changeHandedness) {
-          for (int i = 0; i < usdMesh.points.Length; i++) {
-            usdMesh.points[i] = UnityTypeConverter.ChangeBasis(usdMesh.points[i]);
+      // TODO: indices should not be accessed if topology is not requested, however it may be
+      // needed for facevarying primvars; that special case should throw a warning, rather than
+      // reading the value.
+      Vector3[] newVertices = null;
+      int[] newIndices = null;
+      bool arePointsSanitized = false;
+
+      if (options.meshOptions.points == ImportMode.Import && usdMesh.points != null)
+      {
+        if (changeHandedness)
+        {
+          newVertices = new Vector3[usdMesh.points.Length];
+          for (int i = 0; i < usdMesh.points.Length; i++)
+          {
+            newVertices[i] = UnityTypeConverter.ChangeBasis(usdMesh.points[i]);
           }
         }
+        
+        if (IsThereFaceVaryingAttributes(usdMesh, options))
+        {
+          newVertices = SanitizePointsForUnity(newVertices ?? usdMesh.points, usdMesh.faceVertexIndices);
```

Comment:
I would add a comment here explaining what's happening and how it affects the rest of the import process below.

Reply:
Also, the faces only need to be split on seams, thought that is much more of a pain to implement.